### PR TITLE
How to work with Lombok on Quarkus

### LINKED ISSUE #4657: How to work with Lombok on Quarkus
#### Proposed title of the article

How to work with Lombok on Quarkus

#### Proposed article introduction

[Languages] Java language is a simple language for app development. The only turn-off to newbies is that it has so many syntactical terms to cram all at once.
This makes the newcomers confused and also wish for any way to shorten the code for them. Frameworks such as SpringBoot, Micronaut, Quarkus, and others have tried to come up with ingenious ways to do this.
This includes the use of easy-to-use libraries. These libraries provide annotations that can be quickly and easily used to replace blocks of codes.

Lombok is an example of one of these libraries. 'Project Lombok' as it's known, is a java library that plugs into your editor or IDE and builds tools simplifying Java.
It takes away the writing of getters, setters, or equals methods again. This is by use of one annotation.  The class will have a fully-featured builder, Automated logging variables, and much more.

#### Key takeaways

At the end of this article, the following will have been learned:

- How to set up Lombok in a Quarkus project
- IntelliJ lombok setup
- How to set up tests in Quarkus to test the Lombok project functionality
- How to deal with Getters and Setters in Lombok
- Access levels and constructors in Lombok
- EqualsAndHashCode and ToString annotations in Lombok
- Exclude and Data annotations
- Accessors and NotNull
- Builder, BuilderDefault, and Singular annotations in Lombok
- Slf4j, Lombok configurations, and custom logs

#### Article quality

This is an in-depth article on how to use the Lombok library in Quarkus. It also integrates the use of tests to the code to prove the functionality of the library.

#### References

- [Lombok source code](https://github.com/projectlombok/lombok)

### Templates to use as guides

- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by RisoriTofa**
#### Formatting and structure checks

- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **See [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites? See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and has the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks

- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks

- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #4657

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✔️

**Comment by RisoriTofa** [comment]
@jerimkaura all is good to move on!!!

**Comment by RisoriTofa** [comment]
@jerimkaura, am waiting for any pings

**Review by jerimkaura — CHANGES_REQUESTED** [review]
Hello @RisoriTofa :wave:

Thank you for such an in-depth article. Such content is great for our readers. :rocket: However there are age issues that if addressed will improve the overall quality of this article and other articles as well

- Organise your ideas into explained concepts given as complete sensible paragraphs. Learn to separate where to use bulleted instructions and sentence explanations.
- Keywords, variables, folder names are enclosed in backticks. I recommend you use that instead of `*`. Do this across the whole article.
- HIghlight most annotations types/names discussed in the article.

Work on these then ping me to proceed with the review.

**Comment by RisoriTofa** [comment]
> Hello @RisoriTofa wave
> 
> Thank you for such an in-depth article. Such content is great for our readers. rocket However there are age issues that if addressed will improve the overall quality of this article and other articles as well
> 
> * Organise your ideas into explained concepts given as complete sensible paragraphs. Learn to separate where to use bulleted instructions and sentence explanations.
> * Keywords, variables, folder names are enclosed in backticks. I recommend you use that instead of `*`. Do this across the whole article.
> * HIghlight most annotations types/names discussed in the article.
> 
> Work on these then ping me to proceed with the review.

Done

**Comment by RisoriTofa** [comment]
@jerimkaura are there any updates???

**Comment by jerimkaura** [comment]
Hello @RisoriTofa 

Please be patient as we work on PRs that came into the queue before this. I will communicate if there's anything we need to add/correct.

Thank you!

**Review by jerimkaura — COMMENTED** [review]
Hello, @RisoriTofa 

Kindly go over the changes I have made to ensure you do easy-to-read articles in the future.

Proceeding to tech review.

**Comment by RisoriTofa** [comment]
> Hello, @RisoriTofa
> 
> Kindly go over the changes I have made to ensure you do easy-to-read articles in the future.
> 
> Proceeding to tech review.

I've seen them, and I think they are well thought of... I hope that next time I will use them to improve my articles :fist_oncoming:

**Review by jerimkaura — APPROVED** [review]
Thank you for providing such a deep insight on this topic. However, try to shorten your article next time.

Let us wait for the final review.

**Comment by lalith1403** [comment]
Jumping on this one.

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (lalith1403) [inline content/articles/how-to-work-with-lombok-on-quarkus/index.md:253]: Move this para before the code is introduced.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
There are several readbility issues. The article is long, but needs additional explanation at a lot of places. Please check the comments mentioned. 

Also, make a note of the changes requested, @jerimkaura

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:21]: The extensions are the ones simplifying Java

Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:21]: Let me make the corrections

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:49]: That is supposed to mean that it is installed on the machine... I'm working on that too

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:50]: Done!

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:242]: Ok, got you!

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:253]: Good point!

File: content/articles/how-to-work-with-lombok-on-quarkus/index.md
Comment (RisoriTofa) [reply content/articles/how-to-work-with-lombok-on-quarkus/index.md:49]: Done

**Comment by RisoriTofa** [comment]
@lalith1403 @jerimkaura I have done the changes requested, hope you find them well

**Review by lalith1403 — APPROVED** [review]

**Pull request merged**
